app/src/main/java/com/android/doctorapp/ui/userdashboard/userfragment/UserAppointmentFragment.kt

```diff
@@ -52,11 +70,98 @@ class UserAppointmentFragment :
             lifecycleOwner = viewLifecycleOwner
             viewModel = this@UserAppointmentFragment.viewModel
         }
+
         setUpWithViewModel(viewModel)
+        RuntimePermission.askPermission(
+            this,
+            android.Manifest.permission.ACCESS_COARSE_LOCATION,
+            android.Manifest.permission.ACCESS_FINE_LOCATION
+        ).onAccepted {
+            requestLocationUpdates()
+        }.onDenied {
+            context?.toast(getString(R.string.location_permission))
+        }.ask()
+
         registerObserver(layoutBinding)
         return layoutBinding.root
     }
 
+
+    fun requestLocationUpdates() {
+        val fineLocationPermission = android.Manifest.permission.ACCESS_FINE_LOCATION
+        val coarseLocationPermission = android.Manifest.permission.ACCESS_COARSE_LOCATION
+
+        if (ActivityCompat.checkSelfPermission(
+                requireActivity(),
+                fineLocationPermission
+            ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+                requireActivity(),
+                coarseLocationPermission
+            ) != PackageManager.PERMISSION_GRANTED
+        ) {
+            // Permissions are not granted, request them using the launcher
+            requestLocationPermissionLauncher.launch(
```

Comment:
use like below and remove below line
val fineLocationPermission = android.Manifest.permission.ACCESS_FINE_LOCATION
        val coarseLocationPermission = android.Manifest.permission.ACCESS_COARSE_LOCATION

 requestLocationPermissionLauncher.launch(
                arrayOf(android.Manifest.permission.ACCESS_FINE_LOCATION,android.Manifest.permission.ACCESS_COARSE_LOCATION)
            )